src/components/Components/ComponentDetails/tabs/ComponentDetails.tsx

```diff
@@ -104,7 +110,12 @@ const ComponentDetails: React.FC<React.PropsWithChildren<ComponentDetailsProps>>
                 />
               </DescriptionListTerm>
               <DescriptionListDescription data-test="image-repository-visibility">
-                <ComponentImageRepositoryVisibility component={component} />
+                <ComponentImageRepositoryVisibility
+                  component={component}
+                  imageRepository={imageRepository}
+                  loaded={imageRepoLoaded}
+                  error={imageRepoError}
```

Comment:
just call the useImageRepository hook inside the hook, else use the loading and error state here to not render the `ComponentImageRepositoryVisibility` and just pass the data.